packages/plugin-mocks/src/index.ts

```diff
@@ -37,18 +45,48 @@ export class MocksPlugin<Types extends SchemaTypes> extends BasePlugin<Types> {
     return subscribeMock ?? subscribe;
   }
 
-  resolveMock(typename: string, fieldName: string, mocks: ResolverMap<Types>) {
-    const fieldMock = mocks[typename]?.[fieldName] || null;
+  resolveMock(
+    typeName: string,
+    fieldName: string,
+    outputType: PothosOutputFieldType<Types>,
+    mocks: ResolverMap<Types>,
+    typeMocks: Mock<Types>[],
+  ): Resolver<unknown, {}, Types['Context'], unknown> | null {
+    const fieldMock = mocks[typeName]?.[fieldName] || null;
+
+    if (fieldMock) {
+      if (typeof fieldMock === 'function') {
+        return fieldMock;
+      }
+
+      return fieldMock.resolve ?? null;
+    }
 
-    if (!fieldMock) {
-      return null;
+    if (outputType.kind === 'Object') {
+      const outputName = (outputType.ref as { name: string }).name;
+      const mock = typeMocks.find((v) => v.name === outputName);
+
+      return mock?.resolver ?? null;
     }
 
-    if (typeof fieldMock === 'function') {
-      return fieldMock;
+    if (outputType.kind === 'Interface') {
+      const outputName = (outputType.ref as { name: string }).name;
+      const implementers = this.builder.configStore
+        .getImplementers(outputName)
+        .map((implementer) => implementer.name);
+      const mock = typeMocks.find((v) => implementers.includes(v.name));
+
+      return mock?.resolver ?? null;
```

Comment:
This probably needs a way to make it more configurable....  They way this is written, if you mock any node, any node/nodes query will return that type regardless of what id is being queried. Not sure how to solve this though